angular.js

```diff
@@ -32,5 +32,28 @@ module.exports = {
             Overrides to Angular extra recommended rules:
             https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/configs/recommended--extra.json
         */
+
+        /*
+            Overrides to Angular rules outside of the recommended configuration:
+        */
+
+        '@angular-eslint/component-max-inline-declarations': 'error',
+        '@angular-eslint/component-selector': 'error',
+        '@angular-eslint/contextual-decorator': 'error',
+        '@angular-eslint/directive-selector': 'error',
```

Comment:
[@angular-eslint/directive-selector](http://codelyzer.com/rules/directive-selector)

Reply:
This rule forces a directive to be an element or attribute which I don't think is appropriate in general. It also requires application specific configuration. 

**Proposal**
Leave this rule disabled.

Reply:
We discussed this rule in person and didn't really understand what it enforced regarding element vs attribute. Is it saying that directives can only apply to elements or attributes but not both? If so we agree that seems inappropriate.

Similar to `component-selector`, if it's possible to enforce just `kebab-case` then that would be nice-to-have.

@TrevorKarjanis  can you help us understand what's possible?

Reply:
For reasons described in the rule documentations as well as the [style guide](https://angular.io/guide/styleguide#style-02-06), attribute directives are camel case. Yes, though, this enforces that Directives apply to either elements or attributes of elements, e.g. `app-root` or `app-root[appDirective]`. I'm not sure either is more common, because both are excellent for extending the functionality. Element directives simply apply to all elements of a certain type. Again, this rule doesn't work without a prefix.